docs/realm.js

```diff
@@ -231,6 +231,17 @@ class Realm {
      * @returns {true} if compaction succeeds.
      */
     compact() {}
+
+    /**
+     * If the Realm is a partially synchronized Realm, fetch and synchronize the objects
+     * of a given object type that match the given query (in string format).
+     *
+     * **Partial synchronization is a tech preview. Its APIs are subject to change.**
+     * @param {Realm~ObjectType} type - The type of Realm objects to retrieve.
+     * @param {string} query - Query used to filter objects.
+     * @return {Promise} - a promise that will be resolved with the Realm.Results instance when it's available.
+     */
+    subscribeToObjects(className, query, callback) {}
```

Comment:
why not Realm.Sync.partialSync(className, query) or or if this must be on instance realm.syncSession.partialSync() or realm.syncSession.sync()

subscribeToObjects sounds more like notify for changes on these objects.

Reply:
I simply used the same name as Realm Cocoa and Realm Java.

Reply:
its a poor name for that, but i guess we need to stick with it. is it exposed on the realm there as well? ( I haven't checked at all what other bindings are doing)

Reply:
Yes, it's an instance method on `RLMRealm`.

Reply:
It can’t be on the session as there’d be know way to know which thread’s Realm the results should be associated with.

Reply:
@bdash  I didn't not get that. Isn't realm.syncSession enough to designate the target Realm. I think I didn't understand your comment correctly.

Reply:
A single sync session is shared by all Realm instances for the same sync server URL. It’s necessary to know the specific Realm from which the matching objects should be returned (and also in which Realm we should perform the modifications to register for partial sync).

Reply:
@bdash  It is clear now. thanks. 

@kneth So these suggestions `realm.syncSession` can't work without passing an argument for the realm which will be awkward. `(realm.syncSession.subscribeToObjects(realm...)`

 so I guess either we leave it as it is or we do expose it on Realm.Sync ie `Realm.Sync.subscribeToObjects(realm, className, xx)` this has a better indication that it is related to sync than realm.subscribeToObjects which totally hides the fact, but is a mouthful.

Reply:
I would strongly suggest following the lead of the other bindings for now rather than inventing something different. The API is just a preview and will be revised in the future.

Reply:
indeed staying as it is seems the best approach right now